run.sh

```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -euo pipefail
```

Comment:
I think now even has to work and also instance connect, right?
I think it would be better to not fail on error and start with instance connect and if that does not work fallback to even, or the other way around.